apps/web-giddh/src/app/shared/primary-sidebar/company-branch/company-branch.component.ts

```diff
@@ -0,0 +1,429 @@
+            if (selectedCmp && selectedCmp?.uniqueName === this.generalService.companyUniqueName) {
+                this.selectedCompany = observableOf(selectedCmp);
+                this.activeCompany = selectedCmp;
+                this.companyInitials = this.generalService.getInitialsFromString(selectedCmp.name);
+                this.activeDesign = (this.activeCompany?.name.charCodeAt(0) <= 77) ? 1 : 2;
+
+                if(!this.companyBranches?.branches) {
+                    this.companyBranches = selectedCmp;
+                }
+
+                this.currentCompanyBranches$.subscribe(response => {
+                    if (response && response.length) {
+                        this.branchList = response.sort(this.generalService.sortBranches);
+                        this.currentCompanyBranches = this.branchList;
+                        this.companyBranches.branches = this.branchList;
+                        this.companyBranches.branchCount = this.branchList?.length
+                        this.changeDetectorRef.detectChanges();
+                    }
+                });
+            }
+        });
+
+        this.store.pipe(select(appStore => appStore.session.currentOrganizationDetails), takeUntil(this.destroyed$)).subscribe((organization: Organization) => {
+            this.currentBranchUniqueName = "";
+
+            if (organization && organization.details && organization.details.branchDetails) {
+                this.currentBranchUniqueName = organization.details.branchDetails.uniqueName;
+            }
+        });
+    }
+
+    /**
+     * Releases the occupied memory
+     *
+     * @memberof CompanyBranchComponent
+     */
+    public ngOnDestroy(): void {
+        this.destroyed$.next(true);
+        this.destroyed$.complete();
+    }
+
+    /**
+     * Company filtering method
+     *
+     * @param {*} ev Modal change event
+     * @memberof CompanyBranchComponent
+     */
+    public filterCompanyList(event: any): void {
+        let companies: CompanyResponse[] = [];
+        this.companies$?.pipe(take(1)).subscribe(cmps => companies = cmps);
+
+        this.companyListForFilter = companies?.filter((cmp) => {
+            if (!cmp.alias) {
+                return cmp.name.toLowerCase().includes(event.toLowerCase());
+            } else {
+                return cmp.name.toLowerCase().includes(event.toLowerCase()) || cmp.alias.toLowerCase().includes(event.toLowerCase());
+            }
+        });
+    }
+
+    /**
+     * Change company callback method
+     *
+     * @param {string} selectedCompanyUniqueName Selected company unique name
+     * @param {boolean} [fetchLastState] True, if last state of the company needs to be fetched
+     * @memberof CompanyBranchComponent
+     */
+    public changeCompany(selectedCompanyUniqueName: string, selectBranchUniqueName: string, fetchLastState?: boolean) {
```

Comment:
Please check this method is also present in `primary-sidebar.component.ts`, do remove it from there if unused.